parsers/EIA.py

```diff
@@ -549,6 +575,25 @@ def _fetch(
     ]
 
 
+def _order_by_timestamp(datapoints: List[dict]) -> Dict[str, dict]:
+    indexed_data = {}
+    for datapoint in datapoints:
+        indexed_data[datapoint["datetime"]] = datapoint
```

Comment:
I think that the naming can be improved here. It seems to me that the function does more of an indexing operation than an ordering one.